src/compiler/cbs/VectorShapeTransformer.cpp

```diff
@@ -23,6 +25,17 @@
 using namespace hipsycl::compiler;
 using namespace llvm;
 
+#if LLVM_VERSION_MAJOR < 13
```

Comment:
Aaaaaaa :/ Thanks for the info. Guess we need add a macro to force same treatment as clang < 13 for ROCm :/ FWIW, I believe hipSYCL works out-of-the-box with all functionality with the clang from ROCm 5.2.

Reply:
> FWIW, I believe hipSYCL works out-of-the-box with all functionality with the clang from ROCm 5.2.

Yes, can confirm that. Unfortunately, on our local test machine, 4.5 is the most stable.

> Guess we need add a macro to force same treatment as clang < 13 for ROCm :/ 

I just used `|| defined(HIPSYCL_NO_DEVICE_MANGLER)` here as a workaround, but perhaps it would be easier for users if there was a single `USE_ROCM_CLANG` flag that automatically enables the required workarounds for the current version (`HIPSYCL_NO_DEVICE_MANGLER`, #709, this one here, etc).

Unfortunately, https://github.com/RadeonOpenCompute/ROCm/issues/1680 does not see any action :unamused: